src/app/views/realms/view/overview/RouteRealmsOverview.vue

```diff
@@ -0,0 +1,153 @@
+<script lang="ts">
+import { computed, Ref, ref } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import { Api } from '../../../../../_common/api/api.service';
+import AppButton from '../../../../../_common/button/AppButton.vue';
+import { FiresidePost } from '../../../../../_common/fireside/post/post-model';
+import AppLoadingFade from '../../../../../_common/loading/AppLoadingFade.vue';
+import {
+	createAppRoute,
+	defineAppRouteOptions,
+} from '../../../../../_common/route/route-component';
+import { $gettextInterpolate } from '../../../../../_common/translate/translate.service';
+import { kLineHeightComputed } from '../../../../../_styles/variables';
+import { ActivityFeedService } from '../../../../components/activity/feed/feed-service';
+import { ActivityFeedView } from '../../../../components/activity/feed/view';
+import AppFiresideAvatar from '../../../../components/fireside/avatar/AppFiresideAvatar.vue';
+import AppFiresideAvatarAdd from '../../../../components/fireside/avatar/AppFiresideAvatarAdd.vue';
+import { AppActivityFeedLazy } from '../../../../components/lazy';
+import AppPostAddButton from '../../../../components/post/add-button/AppPostAddButton.vue';
+import { routeRealmsViewFiresides } from '../firesides/firesides.route';
+import { useRealmRouteStore } from '../view.store';
+
+export default {
+	...defineAppRouteOptions({
+		resolver: ({ route }) =>
+			Promise.all([
+				Api.sendRequest('/web/realms/' + route.params.path),
+				Api.sendRequest(
+					ActivityFeedService.makeFeedUrl(
+						route,
+						`/web/posts/fetch/realm/${route.params.path}`
+					)
+				),
+			]),
+	}),
+};
+</script>
+
+<script lang="ts" setup>
+const { realm, firesides, processPayload } = useRealmRouteStore();
+const router = useRouter();
+const route = useRoute();
+
+const feed = ref(null) as Ref<ActivityFeedView | null>;
+const isBootstrapped = ref(false);
+
+// How many firesides to show in the preview row, including the "add a fireside".
+const firesidesGridColumns = 5;
+
+const displayablePreviewFiresides = computed(() => {
+	// -1 to leave room for the "add a fireside"
+	return firesides.value.slice(0, firesidesGridColumns - 1);
+});
```

Comment:
check `src/app/views/communities/view/overview/overview.vue` for how this was handled

Reply:
i did, david commented that we shouldnt use Object.freeze if thats what youre referring to. tbh I don't understand why its doing that anyways, and as for being able to create a fireside - thats not needed. currently you can always create a fireside in any realm, so always leave room for add a fireside.

its the same situation if you already have a fireside running. in this case the -1 would be leaving room for your own fireside.

Reply:
clarification, this is referring to AppHomeFireside's displayFiresides basically.

fix it for both communities route and realms route